api/src/main/java/org/openmrs/module/ssemrreports/reporting/library/data/evaluator/LastEAC1DateDataEvaluator.java

```diff
@@ -37,10 +37,8 @@ public EvaluatedPersonData evaluate(PersonDataDefinition definition, EvaluationC
 	        throws EvaluationException {
 		EvaluatedPersonData c = new EvaluatedPersonData(definition, context);
 		
-		String qry = "SELECT client_id, DATE_FORMAT(max(encounter_datetime), '%Y-%m-%d')  as lastEac1Date FROM ssemr_etl.ssemr_flat_encounter_high_viral_load "
-		        + " WHERE  date(encounter_datetime) <= date(:endDate) "
-		        + " AND eac_session = 'First EAC Session' "
-		        + " GROUP BY client_id;";
+		String qry = "SELECT client_id, DATE_FORMAT(MID(MAX(CONCAT(encounter_datetime, adherence_date)), 20), '%Y-%m-%d') AS lastEac1Date "
```

Comment:
Why did you remove the session filter? Given that the form used for the various sessions is the same, how will you now differentiate between session1, session2, session3 etc?

Reply:
@moshonk we now have different sections for the EAC sessions, where each EAC session has a date field.
![Screenshot from 2024-10-10 02-50-48](https://github.com/user-attachments/assets/7801f0d2-46b8-442a-98e5-2624267f845d)

We removed the dropdowns.

Reply:
Alright @Michaelndula. looks good then.